Armadillo/src/main/java/com/scribd/armadillo/Reducer.kt

```diff
@@ -141,6 +143,19 @@ internal object Reducer {
                         isLoading = true))
                         .apply { debugState = newDebug }
             }
+            is MediaRequestUpdateAction -> {
+                val playbackInfo = oldState.playbackInfo ?: throw ActionBeforeSetup(action)
+                if (playbackInfo.audioPlayable.request.url != action.mediaRequest.url) {
+                    throw InvalidRequest("MediaRequestUpdate cannot be used to change playback URL")
+                }
+                oldState.copy(
+                    playbackInfo = playbackInfo.copy(
+                        audioPlayable = playbackInfo.audioPlayable.copy(
+                            request = action.mediaRequest
+                        )
+                    )
+                ).apply { debugState = newDebug }
```

Comment:
TIL `copy()` doesn't work with non-constructor properties. 😕

Reply:
That's exactly why it's not in the constructor